### ISSUE TITLE
On Export: Duplication of smLink entries

### ISSUE THREAD

**Issue by stefanCCS** [kitodo/kitodo-production#4064]
In meta.xml I have some smLink entries, which are visible correctly in meta data editor.
In my example (nearly) each phyical page has its own logical entry.
After export these smLinks are duplicated a lot.
It looks like that for each logical entry ALL phyisical pages (besides the ones, which are not linked to anything) a smLink-entry is created.
In my opinion this cannot happen during xslt processing. Therefore, this should be something related to Java-Part of export.

**Comment by matthias-ronge** [comment]
That's right, and that's intended. In the internal format, the pages are only assigned to the logical divisions of the respective lowest logical division. So if a page is assigned to a chapter in a book, then it is only assigned to the chapter. It is also assigned to the book during export. Otherwise, no pages would appear in the presentation when you select the book. So far, the behavior is intentional. So I close the issue.

(It would be wrong if an existing `<smLink>` were duplicated, i.e. if it were referred to the same page twice from a logical division. If that is what you were trying to say, it should be corrected. Then please let me know.)

**Comment by matthias-ronge** [comment]
That's right, and that's intended. **In the internal format,** the pages are only assigned to the logical divisions of the lowest logical hierarchy level. So if a page is assigned to a chapter in a book, then it is _only_ assigned to the chapter. It was simpler in terms of programming. **During export,** it is also assigned to the book. Otherwise, no pages would appear in the presentation when you select the book. So far, the behavior is intentional. So I close the issue.

(It would be wrong if an existing `<smLink>` were duplicated, i.e. if it were referred to the same page twice from a logical division. If that is what you were trying to say, it should be corrected. Then please let me know.)

**Comment by matthias-ronge** [comment]
That's right, and that's intended. **In the internal format,** the pages are only assigned to the logical divisions of the lowest logical hierarchy level. So if a page is assigned to a chapter in a book, then it is _only_ assigned to the chapter. It was simpler in terms of programming. **During export,** [it is also assigned](https://github.com/kitodo/kitodo-production/blob/faeecab9f629ae96a966d590cf93ab265fc944f3/Kitodo/src/main/java/org/kitodo/production/services/schema/SchemaService.java#L110-L120) to the book. Otherwise, no pages would appear in the presentation when you select the book. So far, the behavior is intentional. So I close the issue.

(It would be wrong if an existing `<smLink>` were duplicated, i.e. if it were referred to the same page several times from the same logical division. If that is what you were trying to say, it should be corrected. Then please let me know.)

**Comment by matthias-ronge** [comment]
In the internal format, the pages are only assigned to the logical divisions of the lowest logical hierarchy level. So if a page is assigned to a chapter in a book, then it is _only_ assigned to the chapter. It was simpler in terms of programming. During export, [it is also assigned](https://github.com/kitodo/kitodo-production/blob/faeecab9f629ae96a966d590cf93ab265fc944f3/Kitodo/src/main/java/org/kitodo/production/services/schema/SchemaService.java#L110-L120) to the book. Otherwise, no pages would appear in the presentation when you select the book. So far, the behavior is intentional.

However, existing `<smLink>` are duplicated, i.e. from the same logical division it were referred to the same page several times . This needs to be corrected.

**Comment by matthias-ronge** [comment]
[`MetadataEditor`, line 334](https://github.com/kitodo/kitodo-production/blob/master/Kitodo/src/main/java/org/kitodo/production/metadata/MetadataEditor.java#L334) should actually prevent that. Needs checking why it is not doing its job.

**Comment by andre-hohmann** [comment]
@stefanCCS : Could this be solved by the following issue? 
- #5286

**Comment by stefanCCS** [comment]
> @stefanCCS : Could this be solved by the following issue?
> 
>     * [Multiple xlink:to elements for one xlink:from element in mets:structLink #5286](https://github.com/kitodo/kitodo-production/issues/5286)

I need to re-check this.
Will do this with Release 3.5 (which I have not installed so far).

**Comment by BartChris** [comment]
I suspect this was solved by https://github.com/kitodo/kitodo-production/pull/5288 and will close the issue. If the problem is still there, the issue can be re-opened.
